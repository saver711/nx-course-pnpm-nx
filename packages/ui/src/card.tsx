import type { ReactNode, JSX } from 'react';
import Link from 'next/link';

export function Card({
  children,
  href,
}: {
  className?: string;
  title: string;
  children: ReactNode;
  href: string;
}): JSX.Element {
  return (
    <div className="rounded-lg bg-white p-6 shadow-md">
      <Link className="" href={href}>
        <div className="mb-2 text-gray-500">{children}</div>
      </Link>
    </div>
  );
}
