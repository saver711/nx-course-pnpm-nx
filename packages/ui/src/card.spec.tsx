import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import { Card } from './card';

test('ProjectStatus: renders with empty', () => {
  const { baseElement } = render(
    <Card title="Hello" href="/hello">
      <p>This is a test</p>
    </Card>,
  );

  expect(baseElement).toMatchSnapshot();
});
