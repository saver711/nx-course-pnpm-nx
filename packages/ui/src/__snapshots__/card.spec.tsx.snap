// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectStatus: renders with empty 1`] = `
<body>
  <div>
    <div
      class="rounded-lg bg-white p-6 shadow-md"
    >
      <a
        class=""
        href="/hello"
      >
        <div
          class="mb-2 text-gray-500"
        >
          <p>
            This is a test
          </p>
        </div>
      </a>
    </div>
  </div>
</body>
`;
